3.6/debian-9/rootfs/opt/bitnami/scripts/libmongodb.sh

```diff
@@ -489,6 +606,50 @@ mongodb_set_replicasetmode_conf() {
     fi
 }
 
+########################
+# Create the appropriate users
+# Globals:
+#   MONGODB_ROOT_PASSWORD
+# Arguments:
+#   $1 Name of user
+#   $2 Password for user
+#   $3 Name of database (empty for default database)
+# Returns:
+#   None
+#########################
+mongodb_create_user() {
+    local result
+
+    if [[ -n "${3:-}" ]]; then
+        # MongoDB refuses to create users without passwords. It would
+        # return the following otherwise:
+        # Error: couldn't add user: User passwords must not be empty
+        if [[ -n "${2:-}" ]]; then
+            info "Creating user '$1' in database '$3'..."
+
+            result=$(
+                mongodb_execute 'root' "$MONGODB_ROOT_PASSWORD" "" "127.0.0.1" <<EOF
+db.getSiblingDB('$3').createUser({ user: '$1', pwd: '$2', roles: [{role: 'readWrite', db: '$3'}] })
+EOF
+            )
+        else
+            warn "Cannot create user $1, no password provided"
+        fi
+    else
+        if [[ -n "${2:-}" ]]; then
+            info "Creating user '$1' in default database..."
+
+            result=$(
+                mongodb_execute 'root' "$MONGODB_ROOT_PASSWORD" "" "127.0.0.1" <<EOF
+db.getSiblingDB(db.stats().db).createUser({ user: '$1', pwd: '$2', roles: [{role: 'readWrite', db: db.getSiblingDB(db.stats().db).stats().db }] })
+EOF
+            )
+        else
+            warn "Cannot create user $1, no password provided"
+        fi
+    fi
```

Comment:
I think we can simplify this block:
```suggestion
    local -r user="${1:?user is required}"
    local -r password="${2:-}"
    local -r database"${3:-}"
    local query

    if [[ -z "$password" ]]; then
        warn "Cannot create user '$user', no password provided"
        return 0
    fi
    query="db.getSiblingDB('$database').createUser({ user: '$user', pwd: '$password', roles: [{role: 'readWrite', db: '$database'}] })"
    [[ -z "$database" ]] && query="db.getSiblingDB(db.stats().db).createUser({ user: '$user'', pwd: '$password'', roles: [{role: 'readWrite', db: db.getSiblingDB(db.stats().db).stats().db }] })"
    info "Creating user '$user'..."
    mongodb_execute 'root' "$MONGODB_ROOT_PASSWORD" "" "127.0.0.1" <<< "$query"
```

Reply:
Yes you are almost right!

+ I fixed the quick-edit errors on declaring `database` and the double single quotes in the query
+ Most importantly, I kept catching the result of the query in a local `result` to avoid cluttering the logs with the output of Mongo CLI

Changes upcoming